pkg/inngest/inngest/_internal/execution_lib/models.py

```diff
@@ -98,9 +105,16 @@ def __init__(
         self._done_signal = asyncio.Future[None]()
 
     async def __aenter__(self) -> ReportedStep:
+        if _in_step.get() is True:
+            self.info.op = server_lib.Opcode.STEP_ERROR
+            raise step_lib.NestedStepInterrupt()
+        self._in_step_token = _in_step.set(True)
```

Comment:
If the userland code is async, is it possible that we could hit this during a step execution without a step being nested?